misc/userscripts/qute-keepass

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python3
+
+# Copyright 2018 Jay Kamat <jaygkamat@gmail.com>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+# pylint: disable=bad-builtin
+
+USAGE = """This userscript allows for insertion of usernames and passwords from
+keepass databases using pykeepass. Since it is a userscript, it must be run from
+qutebrowser.
+
+A sample invocation of this script is:
+
+:spawn --userscript qute-keepass -p ~/KeePassFiles/MainDatabase.kdbx
+
+And a sample binding
+
+:bind --mode=insert <ctrl-i> spawn --userscript qute-keepass -p ~/KeePassFiles/MainDatabase.kdbx
+
+-p or --path is a required argument.
+
+--keyfile-path allows you to specify a keepass keyfile. If you only use a
```

Comment:
Why bother having a `--no-password` argument which is mandatory with `--keyfile-path` and forbidden without it? Why not simply have `--keyfile-path`?

Reply:
Er, there are 3 states that we can have:

- Password Only
- Keyfile and Password
- Keyfile Only

If you only have a keyfile, you'll need to specify `--no-password` to prevent a password prompt. I think most people are in the first category, which is why I made the flags structured in this way.